### PR TITLE
Keep coherence between illustrations of GridSearchCV

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
This PR:

- ~~moves the [illustration of GridSearchCV](https://github.com/INRIA/scikit-learn-mooc/blob/main/figures/cross_validation_train_test_diagram.png) to the [first GridSearchCV notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_grid_search.html)~~;
- ~~re-works the progression of the concepts to have a more linear narrative~~;
- improves the wording and fixes a typo.

Discussing with @ogrisel we thought that this one can be updated in FUN even if new users start having a slightly different version, but I would rather have the @glemaitre and @lesteve opinions.

**Edit**: #752 was fixed by #847, as it was merged before the current PR. It's scope then changed to improve the wording for better coherence between the [Hyperparameter tuning by grid-search](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_grid_search.html) and the [Evaluation and hyperparameter tuning](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_nested.html) notebooks.

### CODE REVIEW

File: python_scripts/parameter_tuning_nested.py [hunk 1]
```diff
@@ -190,33 +190,14 @@
...
-# Then a final model tuned with those hyper-parameters is fitted on the
-# concatenation of the red and green samples and evaluated on the blue samples.
-#
-# The green samples are sometimes called a **validation sets** to differentiate
-# them from the final test set in blue.
-# ```
+# In the code above, as in some previous notebooks, the selection of the best
+# hyperparameters was done only on the train set from the initial train-test
+# split. Then, we evaluated the generalization performance of our tuned model on
+# the left out test set.
```
Comment (ArturoAmorQ) [inline python_scripts/parameter_tuning_nested.py:196]:
```suggestion
# the left out test set. This can be shown schematically as follows:
#
# ![Cross-validation tuning
# diagram](../figures/cross_validation_train_test_diagram.png)
#
# ```{note}
# This figure shows the particular case of **K-fold** cross-validation
# strategy using `n_splits=5` to further split the train set coming from a
# train-test split.
# For each cross-validation split, the procedure trains a model on all the red
# samples, evaluates the score of a given set of hyperparameters on the green
# samples. The best hyper-parameters are selected based on those intermediate
# scores.
#
# Then a final model tuned with those hyper-parameters is fitted on the
# concatenation of the red and green samples and evaluated on the blue samples.
#
# The green samples are sometimes called a **validation sets** to differentiate
# them from the final test set in blue.
# ```
```

File: python_scripts/parameter_tuning_nested.py [hunk 2]
```diff
@@ -201,16 +202,15 @@
...
-# For each cross-validation split, the procedure trains a model on all the red
-# samples, evaluates the score of a given set of hyperparameters on the green
-# samples. The best hyper-parameters are selected based on those intermediate
-# scores.
+#
+# For each cross-validation split, the procedure trains a model with a given set
+# of hyperparameters on all the red samples, evaluates the score of such
+# combination on the green samples, and selects the best combination of
+# hyperparameters `best_params` by comparing the mean of those validation
+# scores for different combinations.
```
Comment (SebastienMelo) [inline python_scripts/parameter_tuning_nested.py:210]:
```suggestion
# For each cross-validation split, the procedure trains a model with a given set
# of hyperparameters on all the red samples and evaluates the score of such
# combination on the green samples. The best combination of
# hyperparameters `best_params` is selected by comparing the mean of those validation
# scores for different combinations.
```
